### PR TITLE
Adds the ability to request a single node or store status.

### PR DESCRIPTION
**Pull request by BramGruneir**

### CODE REVIEW

**Comment by spencerkimball** [comment]
LGTM

File: server/status.go [hunk 1]
```diff
@@ -208,10 +209,37 @@ func (s *statusServer) handleNodesStatus(w http.ResponseWriter, r *http.Request,
...
+
+	nodeStatus := &proto.NodeStatus{}
+	call := client.GetProto(key, nodeStatus)
+	resp := call.Reply.(*proto.GetResponse)
+	if err := s.db.Run(call); err != nil {
+		log.Error(err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	if resp.Error != nil {
```
Comment (tamird) [inline server/status.go]: is this clause equivalent to the one immediately preceding it? i.e. `resp.Error != nil == err != nil`?

File: server/status.go [hunk 2]
```diff
@@ -254,10 +282,37 @@ func (s *statusServer) handleStoresStatus(w http.ResponseWriter, r *http.Request
...
+
+	storeStatus := &proto.StoreStatus{}
+	call := client.GetProto(key, storeStatus)
+	resp := call.Reply.(*proto.GetResponse)
+	if err := s.db.Run(call); err != nil {
+		log.Error(err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	if resp.Error != nil {
```
Comment (tamird) [inline server/status.go]: same question

File: server/status.go (hunk 1 above)
Comment (tschottdorf) [inline server/status.go]: I think normally yes, but not necessarily. A call can (and `GetProto` does) have a post-handler attached to it, and if everything goes smoothly but that handler fails then you get the result from `db.Run`, but the `resp.Error` is nil.

Comment (tschottdorf) [inline server/status.go]: I do think that checking for `resp.Error` here is pointless though. If that isn't ok, then the whole call won't. I could be overlooking something.

Comment (BramGruneir) [inline server/status.go]: Two different error conditions.  
The first one err != nil is to ensure the communication with the server is correct.
The second one resp.Error will contain an error if there was an problem with the call itself (that will not be present in the err from db.Run.
I found this out the hard way when I thought I had a passing test at one point.

Comment (tschottdorf) [inline server/status.go]: You're right. I was looking at the wrong code in `client/kv.go#Run`.

Comment (tschottdorf) [inline server/status.go]: Or maybe not.
in the snippet below from `Run`, `err` is a named return. That seems to corroborate my initial comment.
```
		kv.Sender.Send(c)
		err = c.Reply.Header().GoError()
		if err != nil {
			log.Infof("failed %s: %s", c.Method(), err)
		} else if c.Post != nil {
			err = c.Post()
		}
		return
```

**Pull request merged**